internal/sign/job/signer.go

```diff
@@ -0,0 +1,138 @@
+}
+
+type signer struct {
+	helper sign.Helper
+	scheme *runtime.Scheme
+}
+
+func NewSigner(helper sign.Helper, scheme *runtime.Scheme) Signer {
+	return &signer{helper: helper, scheme: scheme}
+}
+
+func (m *signer) MakeJobTemplate(mod kmmv1beta1.Module, signConfig *kmmv1beta1.Sign, targetKernel string, previousImage string, targetImage string, pushImage bool) (*batchv1.Job, error) {
+	var args []string
+
+	if pushImage {
+		args = []string{"-signedimage", targetImage}
+	} else {
+		args = append(args, "-no-push")
+	}
+
+	if previousImage != "" {
+		args = append(args, "-unsignedimage", previousImage)
+	} else if signConfig.UnsignedImage != "" {
+		args = append(args, "-unsignedimage", signConfig.UnsignedImage)
+	} else {
+		return nil, fmt.Errorf("no image to sign given")
+	}
+	args = append(args, "-key", "/signingkey/key.priv")
+	args = append(args, "-cert", "/signingcert/public.der")
+
+	if len(signConfig.FilesToSign) > 0 {
+		args = append(args, "-filestosign", strings.Join(signConfig.FilesToSign, ":"))
+	}
+
+	volumes := []v1.Volume{}
+	volumeMounts := []v1.VolumeMount{}
+	volumes = append(volumes, m.makeImageSigningSecretVolume(signConfig.KeySecret, "key", "key.priv"))
+	volumes = append(volumes, m.makeImageSigningSecretVolume(signConfig.CertSecret, "cert", "public.der"))
+
+	volumeMounts = append(volumeMounts, m.makeImageSecretVolumeMount(signConfig.CertSecret, "/signingcert"))
+	volumeMounts = append(volumeMounts, m.makeImageSecretVolumeMount(signConfig.KeySecret, "/signingkey"))
+
+	if mod.Spec.ImageRepoSecret != nil {
+		args = append(args, "-pullsecret", "/docker_config/config.json")
+		volumes = append(volumes, m.makeImageSigningSecretVolume(mod.Spec.ImageRepoSecret, v1.DockerConfigJsonKey, "config.json"))
+		volumeMounts = append(volumeMounts, m.makeImageSecretVolumeMount(mod.Spec.ImageRepoSecret, "/docker_config"))
+	}
+
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: mod.Name + "-sign-",
+			Namespace:    mod.Namespace,
+			Labels:       labels(mod, targetKernel),
+		},
+		Spec: batchv1.JobSpec{
+			Completions: pointer.Int32(1),
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:         "signimage",
+							Image:        "quay.io/chrisp262/kmod-signer:latest",
+							Args:         args,
+							VolumeMounts: volumeMounts,
+						},
+					},
+					RestartPolicy: v1.RestartPolicyOnFailure,
+					Volumes:       volumes,
+					NodeSelector:  mod.Spec.Selector,
+				},
+			},
+		},
+	}
+
+	if err := controllerutil.SetControllerReference(&mod, job, m.scheme); err != nil {
+		return nil, fmt.Errorf("could not set the owner reference: %v", err)
+	}
+
+	return job, nil
+}
+
+func (m *signer) makeImageSigningSecretVolume(secretRef *v1.LocalObjectReference, key string, path string) v1.Volume {
```

Comment:
Do `makeImageSigningSecretVolume`, `makeImageSecretVolumeMount` and `volumeNameFromSecretRef` need to be `Signer` private methods? I think they could just be standalone functions?

Reply:
I suggest we move all the makeVolume functions to some common package, since they can be used as is ( with changes in parameters) both in job makers for build and sign